src/main/java/seedu/address/ui/DetailsPanel.java

```diff
@@ -1,62 +1,33 @@
 package seedu.address.ui;
 
-import java.util.logging.Logger;
-
 import javafx.collections.ObservableList;
 import javafx.fxml.FXML;
-import javafx.scene.control.Label;
-import javafx.scene.control.ListCell;
-import javafx.scene.control.ListView;
-import javafx.scene.layout.Pane;
 import javafx.scene.layout.Region;
 import javafx.scene.layout.VBox;
 import seedu.address.commons.core.DetailsPanelTab;
-import seedu.address.commons.core.LogsCenter;
 import seedu.address.model.person.Person;
 import seedu.address.model.person.PersonEvent;
 
 public class DetailsPanel extends UiPart<Region> {
-    private static final String FXML = "DetailsPanel.fxml";
-    private final Logger logger = LogsCenter.getLogger(DetailsPanel.class);
-
-    @FXML
-    private VBox upcomingEventsPane;
 
-    @FXML
-    private Label upcomingEventsTitle;
-
-    @FXML
-    private ListView<PersonEvent> upcomingDatesListView;
-
-    @FXML
-    private VBox personDetailsPane;
+    private static final String FXML = "DetailsPanel.fxml";
 
-    @FXML
-    private Label personDetailsTitle;
+    private final PersonDetailsTab personDetailsTab;
+    private final UpcomingEventsTab upcomingEventsTab;
 
     @FXML
-    private ListView<Person> personDetailsListView;
+    private VBox tabPlaceholder;
 
     /**
      * Creates a {@code DetailsPanel} with the given {@code ObservableList}s.
-     * @param upcomingDatesList A list of upcoming dates.
+     * @param upcomingEventsList A list of upcoming events.
      * @param detailedPerson A list containing a single {@code Person}.
```

Comment:
I think would be good to document something the rationale for this

Reply:
I realised in some places I named it `UpcomingDates` while in others they were `UpcomingEvents`. Actually, I just realised in `logic` it's still called `UpcomingDates`. I think I'll just revert and switch it all to `UpcomingDates`.

Reply:
Ok, btw I was talking about the usage of a single `Person` in the `detailedPerson` observable list. That one probably need to document.

Reply:
Ok will do. Anyway I think i'll leave it as `UpcomingEvents` for now. Might rename all in another PR